### ISSUE TITLE
How to make notification on-click detection work ?

### ISSUE THREAD

**Issue by kckunal2612** [evollu/react-native-fcm#737]
1. What version of RN and react-native-fcm are you running? 

- "react-native": "0.50.3",
- "react-native-fcm": "11.1.0"

2. What device are you using? (e.g iOS9 emulator, Android 6 device)? **Android 6 Emulator**.

3. Is your app running in foreground, background or not running?  **Background.**

I have followed the code that you have given on the react-native fcm homepage where I have put the FCM.Notification listener outside component scope -

```
// this shall be called regardless of app state: running, background or not running. Won't be called when app is killed by user in iOS

FCM.on(FCMEvent.Notification, async (notif) => {
    // there are two parts of notif. notif.notification contains the notification payload, notif.data contains data payload
    if(notif.local_notification){
        //this is a local notification
        console.log("local notification");
    }
    if(notif.opened_from_tray){
        //iOS: app is open/resumed because user clicked banner
        //Android: app is open/resumed because user clicked banner or tapped app icon
        console.log("opened_from_tray");
    }
....
```
However, when I receive notification from _FCM Notification Composer_ and click on the received notification banner, `notif.local_notification` is not called. Could you help me with how to work with this ? 
I just need to set a small flag when user clicks on notification.

**Comment by evollu** [comment]
`notif.local_notification` is not called?
do you mean the `async (notif)` callback doesn't trigger at all?

**Comment by kckunal2612** [comment]
Sorry, I meant that `notif.opened_from_tray` isn't being invoked.
Also, yeah the `async(notif) `isn't triggering at all. I wanted to see the contents of my notification object, but I am not getting it in the logs.

**Comment by evollu** [comment]
I'm confused here. `receive notification from FCM Notification Composer` so this is a remote notification? then local_notification will be false.
but opened_from_tray should be true since you opened when app is in background

**Comment by kckunal2612** [comment]
Yes it was a remote notification which I sent using my FCM Notification Composer on the Firebase Console. So `local_notification` should be false. However, when I click on the notification, `opened_from_tray` isn't being set to true somehow, which is why I am unable to redirect the user to a custom screen instead of default app landing page.

**Comment by evollu** [comment]
`which is why I am unable to redirect the user to a custom screen instead of default app landing page.`

you mentioned default landing page. so the app was kill state? not background?

**Comment by kckunal2612** [comment]
Yes the app was in the background. I did not forcefully kill the app via recents menu.

**Comment by evollu** [comment]
have no clue here. Can you provide a repo to reproduce it?
Can you reproduce with sample app?
